### PR TITLE
Show cosmosdb nosql resources in azure tree

### PR DESCRIPTION
**Pull request by languy**
Add support for Cosmos DB NoSql resources in the New Connection dialog Azure tree.

image

### CODE REVIEW

**Comment by cheenamalhotra** [comment]
Could you please add gifs/screenshots to the PR description to where you're able to connect to these resources from Azure tree and Browse > Azure tree views?

**Comment by languy** [comment]
> Could you please add gifs/screenshots to the PR description to where you're able to connect to these resources from Azure tree and Browse > Azure tree views?

Sure, I have added a screenshot (redacted to remove resources names).

File: extensions/azurecore/src/azureResource/providers/cosmosdb/nosql/cosmosDbNoSqlTreeDataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { AzureResourceItemType, AzureResourcePrefixes, cosmosDBNoSqlProvider } from '../../../constants';
+import { generateGuid } from '../../../utils';
+import { DbServerGraphData, GraphData } from '../../../interfaces';
+import { ResourceTreeDataProviderBase } from '../../resourceTreeDataProviderBase';
+import { AzureAccountProperties, azureResource } from 'azurecore';
+import * as azdata from 'azdata';
+
+export class CosmosDbNoSqlTreeDataProvider extends ResourceTreeDataProviderBase<GraphData, DbServerGraphData> {
+	private static readonly CONTAINER_ID = 'azure.resource.providers.databaseServer.treeDataProvider.cosmosDbNoSqlContainer';
+	private static readonly CONTAINER_LABEL = localize('azure.resource.providers.databaseServer.treeDataProvider.cosmosDbNoSqlContainerLabel', "CosmosDB NoSql API");
```
Comment (cheenamalhotra) [inline extensions/azurecore/src/azureResource/providers/cosmosdb/nosql/cosmosDbNoSqlTreeDataProvider.ts:19]:
```suggestion
	private static readonly CONTAINER_LABEL = localize('azure.resource.providers.databaseServer.treeDataProvider.cosmosDbNoSqlContainerLabel', "Azure CosmosDB for NoSql");
```

Please follow official naming, as also done for MongoDB: https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/

**Review by cheenamalhotra — APPROVED** [review]
Just a small change.. otherwise LGTM.

File: extensions/azurecore/src/azureResource/providers/cosmosdb/nosql/cosmosDbNoSqlTreeDataProvider.ts (hunk 1 above)
Comment (languy) [reply extensions/azurecore/src/azureResource/providers/cosmosdb/nosql/cosmosDbNoSqlTreeDataProvider.ts]: Good catch! I have also corrected the label string to make it consistent with what you did for Mongo (display the subscription as well).

**Pull request merged**